libs/ml/tests/ml/optimisation/optimisers.cpp

```diff
@@ -118,29 +123,30 @@ TYPED_TEST(OptimisersTest, sgd_optimiser_training)
 {
   using DataType = typename TypeParam::Type;
 
-  DataType learning_rate = DataType{0.1f};
+  DataType learning_rate = DataType{0.05f};
 
   // Prepare model
   std::string                                  input_name;
+  std::string                                  label_name;
   std::string                                  output_name;
   std::shared_ptr<fetch::ml::Graph<TypeParam>> g =
-      PrepareTestGraph<TypeParam>(1, 1, input_name, output_name);
+      PrepareTestGraph<TypeParam>(1, 1, input_name, label_name, output_name);
 
   // Prepare data and labels
   TypeParam data;
   TypeParam gt;
   PrepareTestDataAndLabels1D(data, gt);
 
   // Initialize Optimiser
-  fetch::ml::optimisers::SGDOptimiser<TypeParam, fetch::ml::ops::MeanSquareError<TypeParam>>
-      optimiser(g, {input_name}, output_name, learning_rate);
+  fetch::ml::optimisers::SGDOptimiser<TypeParam> optimiser(g, {input_name}, label_name, output_name,
+                                                           learning_rate);
 
   // Do 2 optimiser steps
   optimiser.Run({data}, gt);
   DataType loss = optimiser.Run({data}, gt);
 
   // Test loss
-  EXPECT_NEAR(static_cast<double>(loss), 0.459031165, 1e-5);
+  EXPECT_NEAR(static_cast<double>(loss), 0.9180623, 1e-5);
```

Comment:
why has this changed?

Reply:
Because MeanSquareError used to be just SquareError divided by 2 instead of SquareError divided by size.